src/app/components/bookmark-menu/bookmark-menu.component.ts

```diff
@@ -45,12 +46,22 @@ export class BookmarkMenuComponent implements OnInit {
     this.icon = this.icon ? this.icon : 'view_list';
     this.nbTopBookmarks = this.nbTopBookmarks ? this.nbTopBookmarks : 3;
 
-    this.bookmarkService.dataBase.dataChange.subscribe(bookmarks => {
-      const sortedBookmark = bookmarks.sort((a, b) => {
-        return (a.views < b.views ? -1 : 1) * (-1);
-      });
-      this.topBookmarks = sortedBookmark.slice(0, this.nbTopBookmarks);
-    });
+    if (this.bookmarkService.dataBase instanceof BookmarkPersistenceDatabase) {
+      (this.bookmarkService.dataBase as BookmarkPersistenceDatabase).dataChange
+        .subscribe((bookmarks: { total: number, items: BookMark[] }) => {
+          const sortedBookmark = bookmarks.items.sort((a, b) => {
```

Comment:
There seems to be the same code in both cases `const sortedBookmark = ...`

Reply:
It's the iterator which is different